client/my-sites/upgrades/components/domain-warnings/index.jsx

```diff
@@ -312,15 +317,20 @@ export default React.createClass( {
 				key="expiring-domains-cannot-manage"
 				text={ text } />
 		);
-	},
+	}
 
 	newDomains() {
+		if ( get( this.props, 'selectedSite.options.is_domain_only' ) ) {
```

Comment:
This is the only "real" change, introduced in a separate commit to ease reviewing: https://github.com/Automattic/wp-calypso/pull/13389/commits/7d71d9a0f12552cfdb0aa321605914c5f17480b8